MastodonSDK/Sources/MastodonSDK/API/Mastodon+API+Account+Friendship.swift

```diff
@@ -215,6 +215,53 @@ extension Mastodon.API.Account {
 
 }
 
+public extension Mastodon.API.Account {
+ 
+    static func blocksEndpointURL(domain: String) -> URL {
+        return Mastodon.API.endpointURL(domain: domain).appendingPathComponent("blocks")
+    }
+    
+    /// Block
+    ///
+    /// Block the given account. Clients should filter statuses from this account if received (e.g. due to a boost in the Home timeline).
+    ///
+    /// - Since: 0.0.0
+    /// - Version: 3.3.0
+    /// # Last Update
+    ///   2021/4/1
+    /// # Reference
+    ///   [Document](https://docs.joinmastodon.org/methods/blocks/)
+    /// - Parameters:
+    ///   - session: `URLSession`
+    ///   - domain: Mastodon instance domain. e.g. "example.com"
+    ///   - authorization: User token.
+    /// - Returns: `AnyPublisher` contains `Relationship` nested in the response
+    static func blocks(
+        session: URLSession,
+        domain: String,
+        authorization: Mastodon.API.OAuth.Authorization
+    ) -> AnyPublisher<Mastodon.Response.Content<[Mastodon.Entity.Account]>, Error>  {
+        let request = Mastodon.API.get(
+            url: blocksEndpointURL(domain: domain),
+            query: BlocksQuery(),
+            authorization: authorization
+        )
+        return session.dataTaskPublisher(for: request)
+            .tryMap { data, response in
+                let value = try Mastodon.API.decode(type: [Mastodon.Entity.Account].self, from: data, response: response)
+                return Mastodon.Response.Content(value: value, response: response)
+            }
+            .eraseToAnyPublisher()
+    }
+    
+    private struct BlocksQuery: GetQuery {
+        var queryItems: [URLQueryItem]? {
+            [URLQueryItem(name: "limit", value: "-1")]
```

Comment:
I haven't tried this API. And could we grab all block users when setting the limit to "-1"?

Reply:
Oh no, this was just for testing purposes and I'll remove it, the backend actually converts it to an absolute number, so we have a limit of 1 in this case. To have a proper implementation we probably need to implement pagination here.